### ISSUE TITLE
Wrong property json when using $ref and @var annotation

### ISSUE THREAD

**Issue by petr-buchin** [zircote/swagger-php#499]
I have a definition for a date fields:
```php
/**
 * @SWG\Definition(
 *     definition="date",
 *     type="integer",
 *     description="Unix timestamp",
 *     example=1519389974
 * )
 */
```

I use it like follows:

```php
    /**
     * @var DateTimeInterface
     * @SWG\Property(ref="#/definitions/date")
     */
    private $createdAt;
```

Normally, this lib should generate such JSON for this field:

```json
"createdAt": {
    "$ref": "#/definitions/date"
}
```

But it takes `@var` annotation into account and generates invalid json for the property:

```json
"createdAt": {
    "type": "string",
    "format": "date-time",
    "$ref": "#/definitions/date"
},
```

By specification, all fields at the same level with `$ref` are ignored, so my Ui generates warnings about this.

Is there any way to ignore `@var` annotation when using refs?

**Comment by bfanger** [comment]
This is indeed a bug, could you create a PR to fix this issue? (Add a check if ref is set in the augmentproperty processor)

**Comment by petr-buchin** [comment]
@bfanger yes, I'll try to create PR this weekend.

**Comment by bfanger** [comment]
This is fixed for v3 in 47cbb3b3145ff46e7b68b4cae903ab8cbd9bbc07
Feel free to backport this change to the 2.x branch
